javascript/HtmlEditorField.js

```diff
@@ -716,10 +716,12 @@ ss.editorWrappers['default'] = ss.editorWrappers.tinyMCE;
 						// http://www.w3.org/TR/1999/REC-html401-19991224/struct/links.html#h-12.2
 
 						if(ed) {
-							var raw = ed.getContent().match(/name="([^"]+?)"|name='([^']+?)'/gim);
+							var raw = ed.getContent()
+								.match(/\s+(name|id)\s*=\s*(["'])([^\2\s>]*?)\2|\s+(name|id)\s*=\s*([^"']+)[\s +>]/gim);
 							if (raw && raw.length) {
 								for(var i = 0; i < raw.length; i++) {
-									collectedAnchors.push(raw[i].substr(6).replace(/"$/, ''));
+									var indexStart = (raw[i].indexOf('id=') == -1) ? 7 : 5;
```

Comment:
Also, does this account for the forms of `id = "something"` and `name = 'something'` etc?